src/App.tsx

```diff
@@ -5,43 +1,33 @@
+import React, { useEffect } from 'react';
+import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
+import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as postsActions from './features/posts';
+import * as selectedPostsActions from './features/selectedPost';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  const dispatch = useAppDispatch();
 
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const { posts, loaded, hasError } = useAppSelector(state => state.posts);
+  const { author } = useAppSelector(state => state.author);
+  const selectedPost = useAppSelector(state => state.selectedPost.selectedPost);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectedPostsActions.setSelectedPost(null));
 
     if (author) {
-      loadUserPosts(author.id);
+      dispatch(postsActions.initPosts(author.id));
     } else {
-      setPosts([]);
+      dispatch(postsActions.setUserPosts([]));
     }
```

Comment:
The conditional logic for dispatching actions based on the `author` state is correct. However, consider adding comments or refactoring to improve readability, especially if this logic becomes more complex in the future.